packages/expo/cli/start/server/middleware/resolveAssets.ts

```diff
@@ -0,0 +1,124 @@
+import { ExpoConfig } from '@expo/config';
+import { BundleAssetWithFileHashes } from '@expo/dev-server';
+import fs from 'fs/promises';
+import path from 'path';
+
+import { getAssetSchemasAsync } from '../../../api/getExpoSchema';
+import * as Log from '../../../log';
+import { fileExistsAsync } from '../../../utils/dir';
+import { CommandError } from '../../../utils/errors';
+import { get, set } from '../../../utils/obj';
+import { validateUrl } from '../../../utils/url';
+
+type ManifestAsset = { fileHashes: string[]; files: string[]; hash: string };
+
+export type Asset = ManifestAsset | BundleAssetWithFileHashes;
+
+type ManifestResolutionError = Error & {
+  localAssetPath?: string;
+  manifestField?: string;
+};
+
+/** Inline the contents of each platform's `googleServicesFile` so runtimes can access them. */
+export async function resolveGoogleServicesFile(
+  projectRoot: string,
+  manifest: Pick<ExpoConfig, 'android' | 'ios'>
+) {
+  if (manifest.android?.googleServicesFile) {
+    try {
+      const contents = await fs.readFile(
+        path.resolve(projectRoot, manifest.android.googleServicesFile),
+        'utf8'
+      );
+      manifest.android.googleServicesFile = contents;
+    } catch (error) {
+      Log.warn(
+        `Could not parse Expo config: android.googleServicesFile: "${manifest.android.googleServicesFile}"`
+      );
+      // Delete the field so Expo Go doesn't attempt to read it.
+      delete manifest.android.googleServicesFile;
+    }
+  }
+  if (manifest.ios?.googleServicesFile) {
+    try {
+      const contents = await fs.readFile(
+        path.resolve(projectRoot, manifest.ios.googleServicesFile),
+        'base64'
```

Comment:
Is this correct for the iOS file to be encoded as ASCII Base64 and the Android file to be UTF-8?

Reply:
https://github.com/expo/expo-cli/blob/bf8edaea7b9bda705ffe0bcc666203bc4879f96b/packages/xdl/src/ProjectAssets.ts#L41-L54